digester.go

```diff
@@ -18,23 +18,39 @@ import (
 	"crypto/sha256"
 	"fmt"
 	hash2 "hash"
+	"reflect"
 	"strings"
 	"sync"
 	"unicode"
+	"unsafe"
 )
 
 // DigestHash generates the digest of statements.
 // it will generate a hash on normalized form of statement text
 // which removes general property of a statement but keeps specific property.
 //
 // for example: both DigestHash('select 1') and DigestHash('select 2') => e1c71d1661ae46e09b7aaec1c390957f0d6260410df4e4bc71b9c8d681021471
+//
+// Deprecated: It is logically consistent with NormalizeDigest.
 func DigestHash(sql string) (result string) {
 	d := digesterPool.Get().(*sqlDigester)
 	result = d.doDigest(sql)
 	digesterPool.Put(d)
 	return
 }
 
+// DigestNormalized generates the digest of a normalized sql.
+// it will generate a hash on a normalized sql.
+// Normalize + DigestNormalized equals to NormalizeDigest.
+//
+// for example: DigestNormalized('select 1') and DigestNormalized('select 2') is not the same
+func DigestNormalized(normalized string) (result string) {
```

Comment:
Why not just `digest`? It does not care about if it is normalized.

Reply:
See https://github.com/pingcap/tidb/issues/13871#issuecomment-561464681 and https://github.com/pingcap/tidb/issues/13871#issuecomment-561516174

Reply:
Oh, so we cannot remove the `DigestHash` or rename it for compatibility reasons? 
The name is quite confusion because it does not say it will normalized the string.

Reply:
Oh, the name of this function is suggested by @lysu , It's intended to by called with a normalized SQL string, otherwise it is meaningless.

Reply:
I've updated the comment of this function.